### PR TITLE
Consuming Multi api Authorization package for stack profile 2017-03-09-profile

### PR DESCRIPTION
**Pull request by bganapa**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by bganapa** [comment]
This seems to have gotten other commits from dev after my rebase, looking at it

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -1215,6 +1215,9 @@ def _create_role_assignment(cli_ctx, role, assignee, resource_group_name=None, s
 
     role_id = _resolve_role_id(role, scope, definitions_client)
     object_id = _resolve_object_id(cli_ctx, assignee) if resolve_assignee else assignee
+    RoleAssignmentCreateParameters = get_sdk(cli_ctx, ResourceType.MGMT_AUTHORIZATION,
+                                             'RoleAssignmentCreateParameters', mod='models',
+                                             operation_group='role_assignments')
```
Comment (troydai) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: @yugangw-msft do we have a better way to centralize this piece?

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py [hunk 2]
```diff
@@ -47,15 +45,40 @@ def list_role_definitions(cmd, name=None, resource_group_name=None, scope=None,
...
+        else:
+            obj[property_name] = property_value
+    else:
+        if hasattr(obj, 'properties'):
+            setattr(obj.properties, property_name, property_value)
+        else:
+            setattr(obj, property_name, property_value)
+
+
+def _create_update_role_definition(cmd, role_definition, for_update):
```
Comment (troydai) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: This method is too long, please break it down into smaller pieces.

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py [hunk 3]
```diff
@@ -81,28 +104,48 @@ def _create_update_role_definition(cli_ctx, role_definition, for_update):
...
         raise CLIError("please provide 'assignableScopes'")
 
+    Permission, RoleDefinition, RoleDefinitionProperties = get_sdk(cmd.cli_ctx, ResourceType.MGMT_AUTHORIZATION,
+                                                                   'Permission', 'RoleDefinition', 'RoleDefinitionProperties', mod='models',
+                                                                   operation_group='role_definitions')
+    if not Permission:
+        raise CLIError('"Permission" configuration is unavailable from the current profile. Please use "az cloud set"'
+                       ' command to use correct one')
+
+    if cmd.supported_api_version(max_api='2015-07-01'):
```
Comment (troydai) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: @yugangw-msft is there a better way?

**Comment by yugangw-msft** [comment]
@bganapa can you please get CLI green by fixing the authorization RP related tests

File: src/command_modules/azure-cli-ams/HISTORY.rst [hunk 4]
```diff
@@ -3,10 +3,18 @@
...
 
+0.2.1
++++++
+* Consuming multi api azure.mgmt.authorization package
+
 0.2.0
 +++++
 * BREAKING CHANGE: 'show' commands log error message and fail with exit code of 3 upon a missing resource.
 
+0.1.2
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-ams/HISTORY.rst]: back out this change

File: src/command_modules/azure-cli-resource/HISTORY.rst [hunk 5]
```diff
@@ -31,7 +31,6 @@ Release History
 ++++++
 * Minor changes
 
-
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-resource/HISTORY.rst]: back out

File: src/command_modules/azure-cli-role/HISTORY.rst [hunk 6]
```diff
@@ -17,6 +21,7 @@ Release History
 
 2.0.25
 ++++++
+* support stack profile
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-role/HISTORY.rst]: back out

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py [hunk 7]
```diff
@@ -81,28 +104,49 @@ def _create_update_role_definition(cli_ctx, role_definition, for_update):
...
     else:
         role_id = _gen_guid()
 
     if not for_update and 'assignableScopes' not in role_definition:
         raise CLIError("please provide 'assignableScopes'")
 
+    Permission, RoleDefinition, RoleDefinitionProperties = get_sdk(cmd.cli_ctx, ResourceType.MGMT_AUTHORIZATION,
+                                                                   'Permission', 'RoleDefinition', 'RoleDefinitionProperties', mod='models',
+                                                                   operation_group='role_definitions')
+    if not Permission:
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: please remove this check as `permission` should always be available, even in older api-version

**Comment by lmazuel** [comment]
@yugangw-msft @bganapa SDL is already released for a few weeks now, this should work without "privates" folder.

File: src/command_modules/azure-cli-ams/HISTORY.rst (hunk 4 above)
Comment (bganapa) [reply src/command_modules/azure-cli-ams/HISTORY.rst]: removed

Comment (bganapa) [reply src/command_modules/azure-cli-ams/HISTORY.rst]: removed

File: src/command_modules/azure-cli-resource/HISTORY.rst (hunk 5 above)
Comment (bganapa) [reply src/command_modules/azure-cli-resource/HISTORY.rst]: removed

File: src/command_modules/azure-cli-role/HISTORY.rst (hunk 6 above)
Comment (bganapa) [reply src/command_modules/azure-cli-role/HISTORY.rst]: removed this line

**Comment by bganapa** [comment]
@yugangw-msft addressed the comments, the tests are green now. I have run the tests for azurestack profile as well and they are all good.  except for @troydai's ones. Can you please suggest how we can fix his comments?

File: src/azure-cli-core/HISTORY.rst [hunk 8]
```diff
@@ -33,10 +37,6 @@ Release History
 ++++++
 * Minor fixes
 
-2.0.36
-++++++
-* Minor fixes
-
```
Comment (yugangw-msft) [inline src/azure-cli-core/HISTORY.rst]: why you remove it? My original comment was about removing redundant entries in your PR

File: src/command_modules/azure-cli-resource/HISTORY.rst [hunk 9]
```diff
@@ -20,10 +20,6 @@ Release History
 ++++++
 * Support deployment at subscription scope: new operation group `deployment`.
 
-2.0.31
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-resource/HISTORY.rst]: Again, why you are removing it?

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py [hunk 10]
```diff
@@ -81,28 +104,50 @@ def _create_update_role_definition(cli_ctx, role_definition, for_update):
...
         role_id = _gen_guid()
 
     if not for_update and 'assignableScopes' not in role_definition:
         raise CLIError("please provide 'assignableScopes'")
 
+    Permission, RoleDefinition, RoleDefinitionProperties = get_sdk(cmd.cli_ctx, ResourceType.MGMT_AUTHORIZATION,
+                                                                   'Permission', 'RoleDefinition',
+                                                                   'RoleDefinitionProperties', mod='models',
+                                                                   operation_group='role_definitions')
+    if not Permission:
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: You don't need to check, as this guy should be available all the time. Please remove this condition

File: src/azure-cli-core/HISTORY.rst (hunk 8 above)
Comment (bganapa) [reply src/azure-cli-core/HISTORY.rst]: oops.. i misunderstood your comment, adding it back

File: src/command_modules/azure-cli-resource/HISTORY.rst (hunk 9 above)
Comment (bganapa) [reply src/command_modules/azure-cli-resource/HISTORY.rst]: adding it back

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py (hunk 10 above)
Comment (bganapa) [reply src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: removed

**Pull request merged**